tests/pamiq_core/threads/test_thread_control.py

```diff
@@ -0,0 +1,266 @@
+import threading
+import time
+
+import pytest
+
+from pamiq_core.threads import (
+    ControllerCommandHandler,
+    ReadOnlyController,
+    ThreadController,
+)
+
+
+class TestThreadController:
+    def test_resume_and_related_predicate_methods(self):
+        thread_controller = ThreadController()
+        thread_controller.resume()
+
+        assert thread_controller.is_resume() is True
+        assert thread_controller.is_pause() is False
+
+    def test_resume_when_shutdown(self):
+        thread_controller = ThreadController()
+        thread_controller.shutdown()
+
+        with pytest.raises(RuntimeError):
+            thread_controller.resume()
+
+    def test_pause_and_related_predicate_methods(self):
+        thread_controller = ThreadController()
+        thread_controller.pause()
+
+        assert thread_controller.is_resume() is False
+        assert thread_controller.is_pause() is True
+
+    def test_pause_when_shutdown(self):
+        thread_controller = ThreadController()
+        thread_controller.shutdown()
+
+        with pytest.raises(RuntimeError):
```

Comment:
このようにするとエラーメッセージまでテストできます。
```suggestion
        with pytest.raises(RuntimeError, match="ThreadController must be activated before pause()."):
```

Reply:
なるほど、ありがとうございます！！！

RuntimeError のテストではメッセージまで見たほうが良いですね、勉強になります！！！